docs/specs/openapi.yaml

```
@@ -238,44 +650,361 @@ paths:
+                        },
+                        "spawn-time": "2024-12-27T10:15:31.390053104+08:00"
+                      }
+                    ],
+                    "ready": false,
+                    "spawn-time": "2024-12-27T10:15:31.390062521+08:00"
+                  }
+                }
+  /v1/changes/{id}/wait:
+    get:
+      summary: Wait for a change to complete
+      description: |
+        Wait for the change to be finished.
+        
+        If the wait operation succeeds, the result will have the "err" field set to an appropriate error message if the change itself had an error.
+      tags:
+        - changes
+      parameters:
+        - in: path
+          name: id
+          schema:
+            type: string
+          required: true
+          description: The ID of the change to wait for.
+        - in: query
+          name: timeout
+          schema:
+            type: string
+          description: |
+            Optional timeout (a [duration](#duration)).
+            If specified, wait till the change is ready or a timeout occurs, whichever is first.
+            If not specified or zero, wait indefinitely until the change is ready.
+      responses:
+        "200":
+          description: Change information
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/GetChangeByIDResponse"
+              example:
+                {
+                  "type": "sync",
+                  "status-code": 200,
+                  "status": "OK",
+                  "result": {
+                    "id": "38",
+                    "kind": "autostart",
+                    "summary": "Autostart service \"svc1\" and 1 more",
+                    "status": "Done",
+                    "tasks": [
+                      {
+                        "id": "54",
+                        "kind": "start",
+                        "summary": "Start service \"svc1\"",
+                        "status": "Done",
+                        "progress": {
+                          "label": "",
+                          "done": 1,
+                          "total": 1
+                        },
+                        "spawn-time": "2024-12-27T12:31:26.673287868+08:00",
+                        "ready-time": "2024-12-27T12:31:27.681780702+08:00"
+                      }
+                    ],
+                    "ready": true,
+                    "spawn-time": "2024-12-27T12:31:26.673297951+08:00",
+                    "ready-time": "2024-12-27T12:31:27.686371869+08:00"
+                  }
+                }
+  /v1/notices:
+    get:
+      summary: Get notices
+      tags:
+        - notices
+      description: Get a list of notices that match the filters, ordered by the last-repeated time.
+      parameters:
+        - in: query
+          name: user-id
+          description: Filter notices by user ID (admin only). Only one user ID can be specified.
```

When I first read this, I thought it meant that the user ID must be the ID of an admin user. How about this wording:
```suggestion
          description: Filter notices by user ID. Only one user ID can be specified. This parameter can only be used by admin users.
```